### PR TITLE
Add more profile types to the profiler

### PR DESCRIPTION
**Pull request by SaveTheRbtz**
This diff:
* creates a basic test for the profiler.
* fixes `block` and `mutex` profiles (they do not work w/o corresponding `SetBlockProfileRate` and `SetMutexProfileFraction`).
* adds support for the `threadcreate` profile.
* improves logging.
* reduces code duplication.

### CODE REVIEW

File: utils/debug/profiler_test.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	t.Parallel()
+	t.Run("profilerEnabled", func(t *testing.T) {
+		tempDir, err := os.MkdirTemp("", "profiles-")
+		require.NoError(t, err)
+		defer os.RemoveAll(tempDir)
+
+		p, err := debug.NewAutoProfiler(log.Logger, tempDir, time.Millisecond*100, time.Millisecond*100, true)
+		require.NoError(t, err)
+
+		<-p.Ready()
```
Comment (jordanschalm) [inline utils/debug/profiler_test.go:24]:
```suggestion
		unittest.AssertClosesBefore(t, p.Ready(), time.Second)
```

File: utils/debug/profiler_test.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+		log.Info().Str("dir", tempDir).Msg("profiler ready")
+
+		require.Eventuallyf(t, func() bool {
+			dirEnts, err := os.ReadDir(tempDir)
+			require.NoError(t, err)
+			return len(dirEnts) > 4
+		}, time.Second*5, time.Millisecond*100, "profiler did not generate any profiles in %v", tempDir)
+
+		<-p.Done()
```
Comment (jordanschalm) [inline utils/debug/profiler_test.go:33]:
```suggestion
		unittest.AssertClosesBefore(t, p.Done(), time.Second)
```

**Review by jordanschalm — APPROVED** [review]
Looks good - thank you for adding test coverage 🎸

**Comment by SaveTheRbtz** [comment]
Thanks!  I'll read that library!

**Comment by SaveTheRbtz** [comment]
^ Just in case you need anything here from automation / production perspective.

**Comment by SaveTheRbtz** [comment]
bors merge

**Pull request merged**